Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-usermenuview', function (Y) {
+    'use strict';
+    /**
+     * Provides the User Menu View class
+     *
+     * @module ez-usermenuview
+     */
+    Y.namespace('eZ');
+
+    var CLASS_HIDDEN = 'is-user-menu-hidden',
+        CLASS_USER_PROFILE = 'ez-user-profile',
+        SELECTOR_USER_PROFILE = '.' + CLASS_USER_PROFILE,
+        SELECTOR_USER_MENU = '.ez-user-menu';
+
+    /**
+     * The User Menu view
+     *
+     * @namespace eZ
+     * @class UserMenuView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UserMenuView = Y.Base.create('userMenuView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-user-menu-item': {
+                'tap': '_toggleUserMenu'
+            },
+            '.ez-user-menu-item-logout': {
+                'tap': '_logOut'
+            }
+        },
+
+        initializer: function () {
+            this.on('*:toggleUserMenu', this._toggleUserMenu);
+        },
+
+        /**
+         * Renders the user menu view
+         *
+         * @method render
+         * @return {eZ.UserMenuView} the view itself
+         */
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template(this.getAttrs()));
+            container.on('clickoutside', this._hideMenuOnClickOutside, this);
+
+            this._renderMenuItems();
+
+            return this;
+        },
+
+        /**
+         * Tap event handler on the logout link
+         *
+         * @method _logOut
+         * @protected
+         * @param event {Object} tap event facade
+         */
+        _logOut: function (event) {
+            event.preventDefault();
+            this.fire('logOut', {
+                originalEvent: event
+            });
+        },
+
+        /**
+         * Toggles the visibility of a user menu.
+         *
+         * @method _toggleUserMenu
+         * @protected
+         */
+        _toggleUserMenu: function () {
+            var methodName = this.get('displayed') ? 'addClass' : 'removeClass',
```

Comment:
this should be separated in 2 steps:

1. `toggleUserMenu` should set the `displayed` attribute value based on the existing one
1. a `displayedChange` event handler should reflect the state in the markup. Also usually, it's easier to set the class on the view container, this avoids to have various selectors in the code, it's easier to unit test and this view the opportunity the style anything in the view (not only the visibility of a specific part).